src/OrchardCore.Modules/Orchard.Autoroute/Handlers/AutoroutePartHandler.cs

```diff
@@ -119,5 +124,29 @@ private string GetPattern(AutoroutePart part)
 
             return pattern;
         }
+
+        private string GenerateUniquePath(string path, AutoroutePart context)
```

Comment:
If you want to extend the validation, create a new class inheriting from `ContentPartDisplayDriver<AutoroutePart>`. This is the extensibility point.

I have a better reason to provide permalink unicity: not everyone is allowed to edit this permalink, it's based on a permission, and if there is a conflict there would be no way for the user to change it. So this is a good reason to add automatic unicity that would make me change my mind. 

So I am ok to keep it.